File: app/src/main/java/io/github/ovso/blackbox/ui/main/di/MainActivityViewModule.java
package io.github.ovso.blackbox.ui.main.di;

import dagger.Binds;
import dagger.Module;
import io.github.ovso.blackbox.ui.main.MainActivity;
import io.github.ovso.blackbox.ui.main.MainPresenter;

@Module public abstract class MainActivityViewModule {

  @Binds abstract MainPresenter.View bindMainView(MainActivity activity);
}

File: app/src/main/java/io/github/ovso/blackbox/ui/main/di/MainActivityViewModule.kt
package io.github.ovso.blackbox.ui.main.di

import dagger.Binds
import dagger.Module
import io.github.ovso.blackbox.ui.main.MainActivity
import io.github.ovso.blackbox.ui.main.MainPresenter

@Module
abstract class MainActivityViewModule {

  @Binds internal abstract fun bindMainView(activity: MainActivity): MainPresenter.View
}